# Capstone - Two: Data Wrangling 

# Predicting Housing Prices in Washington, D.C.

By Gbatchin Kochoni

April, 2025


## Table of Contents

1. Context and Background

2. Project Goal

3. Data Sources

4. Data Wrangling

## 1. Context and Background

Washington, D.C. is a vibrant urban area with a dynamic real estate market. Understanding and predicting property prices can provide valuable insights for buyers, sellers, real estate agents, and urban planners. The project focuses on analyzing the residential property market in D.C. using available public datasets.

## 2. Project Goal

The main objective of this project is to predict the sale prices of residential properties in Washington, D.C. based on property features and characteristics. This prediction model could assist stakeholders in making informed decisions.

Specific goals:

- Analyze patterns and trends in property prices.

- Build a predictive model with high accuracy.

- Identify the most influential features affecting property prices.



## 3. Data Sources

The dataset used is DC_Properties.csv (download from kaggle web site [***]), containing information on residential properties in Washington, D.C., such as:

- Number of rooms, bathrooms, bedrooms.

- Lot area (LANDAREA).

- Property condition, style, structure.

- Sale price (PRICE).

- Location details (ZIPCODE, WARD, QUADRANT).

## 4. Data Wrangling

Key cleaning and preparation steps included:

- Dropping irrelevant columns (e.g., identifiers, coordinates not directly useful).

- Filtering to keep only properties with a positive sale price.

- Removing duplicate entries.

- Handling missing values (planned during feature engineering).

In [8]:

# Import necessary libraries
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns


In [9]:

# Load the dataset
df_properties = pd.read_csv('DC_Properties.csv', low_memory=False)

# Display the first few rows
df_properties.head()


Unnamed: 0  BATHRM  HF_BATHRM           HEAT AC  NUM_UNITS  ROOMS  BEDRM  \
0           0       4          0      Warm Cool  Y        2.0      8      4   
1           1       3          1      Warm Cool  Y        2.0     11      5   
2           2       3          1  Hot Water Rad  Y        2.0      9      5   
3           3       3          1  Hot Water Rad  Y        2.0      8      5   
4           4       2          1      Warm Cool  Y        1.0     11      3   

      AYB  YR_RMDL  ...  LONGITUDE  ASSESSMENT_NBHD ASSESSMENT_SUBNBHD  \
0  1910.0   1988.0  ... -77.040832       Old City 2   040 D Old City 2   
1  1898.0   2007.0  ... -77.040764       Old City 2   040 D Old City 2   
2  1910.0   2009.0  ... -77.040678       Old City 2   040 D Old City 2   
3  1900.0   2003.0  ... -77.040629       Old City 2   040 D Old City 2   
4  1913.0   2012.0  ... -77.039361       Old City 2   040 D Old City 2   

   CENSUS_TRACT CENSUS_BLOCK    WARD  SQUARE          X          Y QUADRANT  
0        4201.0  004201 2006  Ward 2    0152 -77.040429  38.914881       NW  
1        4201.0  004201 2006  Ward 2    0152 -77.040429  38.914881       NW  
2        4201.0  004201 2006  Ward 2    0152 -77.040429  38.914881       NW  
3        4201.0  004201 2006  Ward 2    0152 -77.040429  38.914881       NW  
4        4201.0  004201 2006  Ward 2    0152 -77.040429  38.914881       NW  

[5 rows x 49 columns]


# Data Wrangling
Cleaning and preparing the dataset for analysis.


In [11]:
# Brief summary information about the DataFrame
df_properties.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 158957 entries, 0 to 158956
Data columns (total 49 columns):
 #   Column              Non-Null Count   Dtype  
---  ------              --------------   -----  
 0   Unnamed: 0          158957 non-null  int64  
 1   BATHRM              158957 non-null  int64  
 2   HF_BATHRM           158957 non-null  int64  
 3   HEAT                158957 non-null  object 
 4   AC                  158957 non-null  object 
 5   NUM_UNITS           106696 non-null  float64
 6   ROOMS               158957 non-null  int64  
 7   BEDRM               158957 non-null  int64  
 8   AYB                 158686 non-null  float64
 9   YR_RMDL             80928 non-null   float64
 10  EYB                 158957 non-null  int64  
 11  STORIES             106652 non-null  float64
 12  SALEDATE            132187 non-null  object 
 13  PRICE               98216 non-null   float64
 14  QUALIFIED           158957 non-null  object 
 15  SALE_NUM            158957 non-nul

In [12]:
# Shape of the dataframe befor cleaning
print("Shape befor cleaning:", df_properties.shape)

Shape befor cleaning: (158957, 49)


In [13]:
# Drop unnecessary columns
cols_to_drop = [
    'Unnamed: 0', 'SOURCE', 'CMPLX_NUM', 'GIS_LAST_MOD_DTTM', 
    'SQUARE', 'X', 'Y', 'NATIONALGRID'
]
df = df_properties.drop(columns=cols_to_drop, errors='ignore')

In [14]:
# Filter only rows where price is strictly positive
df = df[df['PRICE'] > 0]

In [15]:
# Remove duplicates
df = df.drop_duplicates()

In [16]:
# Shape of the dataframe after cleaning
print("Shape after cleaning:", df.shape)

Shape after cleaning: (97832, 41)


In [17]:
# Display missing values
missing_values = df.isnull().sum()
missing_values = missing_values[missing_values > 0]
missing_values

NUM_UNITS             39933
AYB                     112
YR_RMDL               40410
STORIES               39966
SALEDATE                  1
GBA                   39933
STYLE                 39933
STRUCT                39933
GRADE                 39933
CNDTN                 39933
EXTWALL               39933
ROOF                  39933
INTWALL               39933
KITCHENS              39934
LIVING_GBA            57899
FULLADDRESS           40267
CITY                  40263
STATE                 40263
ASSESSMENT_SUBNBHD    20039
CENSUS_BLOCK          40263
QUADRANT                101
dtype: int64

**Data wrangling is completed. We have:**
- Dimensions after cleaning: 97,832 rows × 41 columns.
- The selling price (PRICE) is always positive.
- Columns with missing values: Yes, approximately 20 to 40% in some columns such as NUMBER_UNITS, QUALITY, STRUCTURE, LIVING_AREA, etc.

**The dataset is now ready for exploration in the next steps, although we recognize that some columns still have missing values, which can be addressed during the EDA and/or Feature Engineering phases.**